src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -73,6 +84,11 @@ private void StopRaisingEvents()
                 token.Cancel();
                 token.Dispose();
             }
+
+            if (_cachedRunningInstance.IsAllocated)
+            {
+                _cachedRunningInstance.Free();
```

Comment:
Do we want it freed in StopRaisingEvents, or should it be freed only on FinalizeDispose?

Reply:
If it is held by RunningInstance then it should presumably be freed in the Dispose of RunningInstance. And you call Dispose here unless you want to preserve the RunningInstance across Start/Stop calls. Which depends on the semantics of FSW. I'm guessing the semantics are that we release all native resources when you do Stop, in case it holds a handle to the directory being watched for example. So in that case StartRaisingEVents creates a RI and StopRaisingEvents disposes the RI

Reply:
and of course Dispose should also release it in case StopRaisingEvents never got called.